src/sctp_transport/mod.rs

```diff
@@ -212,12 +212,22 @@ impl RTCSctpTransport {
     }
 
     async fn accept_data_channels(param: AcceptDataChannelParams) {
+        let dcs = param.data_channels.lock().await;
+        let mut existing_data_channels = Vec::new();
+        for dc in dcs.iter() {
```

Comment:
Also one idea I just had is to modify `DataChannel::accept` to accept a reference to a list of existing data channels. That way we won't need to clone all of them, just the one with matching identifier.

Reply:
Opened https://github.com/webrtc-rs/data/pull/15